sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/interface.scala

```diff
@@ -198,6 +200,10 @@ case class CatalogTable(
       locationUri, inputFormat, outputFormat, serde, compressed, properties))
   }
 
+  def withStats(cboStatsEnabled: Boolean): CatalogTable = {
```

Comment:
Yes I also think that's better, but as @cloud-fan said, we can't get the config in `def statistics`, we have to modify many places to support this. I'm about to do such modifications, do you have any advices to minimize the changes?